File: Source/Languages/Python/Samples/fibo.py
import time

def Fibo(n):
  if n <= 1 : return n
  else      : return Fibo(n-1) + Fibo(n-2)

def Benchmark(name, func, arg=0, loops=1):
  elapsed = time.time()
  for count in range(loops):
    result = func(arg)
  elapsed = (time.time() - elapsed) / loops
  print("{}({}) = {} in {} seconds".format(name, arg, result, elapsed))

Benchmark("Fibo", Fibo, 24, 10) # Fibo(24) = 46368

File: Source/Testing/Minimal/Fibo.hs
program Fibo
{
    //#define MCU

    uses "/Source/Minimal/IO"
      
    delegate uint FiboDelegate(uint n);
    
    uint Fibo(uint n)
    {
        if ( n <= 1 ) 
        { return n; }
        else
        { return Fibo(n-1) + Fibo(n-2); }
    }
    Benchmark(string name, FiboDelegate func, uint arg, uint loops)
    {
        uint start = Time.Seconds;
        uint result = 0;
        for(uint count=0; count < loops; count++)
        {
            result = func(arg);
        }
        uint elapsed = Time.Seconds - start;
        WriteLn();
        WriteLn(name + " (" + arg.ToString() + ") = " + result.ToString() + " in " + elapsed.ToString() + " seconds");
    }
    Hopper()
    {
        FiboDelegate fiboDelegate = Fibo;
        // 1, 1, 2, 3, 5, 8, 13, 21, 34, 55
        Benchmark("Fibo", fiboDelegate, 24, 1); // Fibo(24) = 46368
    }
}
